### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by linaz678**

### CODE REVIEW

**Review by menjunyi — COMMENTED** [review]
lgtm

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+RUN yarn install
+#To bundle your app's source code inside the Docker image, use the COPY instruction:
+COPY . .
+# should I have yarn build at this stagge ?
+
+RUN ls -a
+
+CMD ["node", "dist/server.js"]
```
Comment (menjunyi) [inline Dockerfile:16]: reviewed

**Review by menjunyi — APPROVED** [review]
lgtm

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Install dependencies only when needed
+FROM node:16-alpine 
+#Next we create a directory to hold the application code inside the image
+WORKDIR /app
+# Install dependencies based on the preferred package manager
+COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
```
Comment (Qiming-Liu) [inline Dockerfile:6]: Why not just `copy . .` here?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+
+RUN yarn install
+#To bundle your app's source code inside the Docker image, use the COPY instruction:
+COPY . .
+# should I have yarn build at this stagge ?
```
Comment (Qiming-Liu) [inline Dockerfile:12]: Correct, should `yarn build` here

File: Dockerfile (hunk 1 above)
Comment (Qiming-Liu) [reply Dockerfile:16]: This is actually a wrong Dockerfile

**Comment by Qiming-Liu** [comment]
Should be
```
FROM node:16-alpine 
WORKDIR /app
COPY . .
RUN yarn && yarn build
CMD ["node", "dist/server.js"]
```

**Pull request merged**